### ISSUE TITLE
Export-O365Configuration Not Prompting for Destination Path

### ISSUE THREAD

**Issue by adrianwells** [microsoft/Microsoft365DSC#183]
#### Details of the scenario you tried and the problem that is occurring
`Export-O365Configuration` does not prompt to enter a Destination Path for the output

#### Verbose logs showing the problem
No logs to show yet. If there's a recommend method to collect them, please advise.

#### Suggested solution to the issue
Work around appears to be `Install-Module MSCloudLoginAssistant -Force -AllowClobber` and try again as noted in the comments here https://nikcharlebois.com/getting-started-with-office365dsc/

#### The operating system the target node is running

OsName               : Microsoft Windows 10 Enterprise
OsOperatingSystemSKU : EnterpriseEdition
OsArchitecture       : 64-bit
WindowsVersion       : 1903
WindowsBuildLabEx    : 18362.1.amd64fre.19h1_release.190318-1202
OsLanguage           : en-US
OsMuiLanguages       : {en-US}

#### Version of the DSC module that was used ('dev' if using current dev branch)
1.0.0.846  Office365DSC

**Comment by NikCharlebois** [comment]
@adrianwells can you confirm that this has been fixed in the current Dev branch? Thanks

**Comment by adrianwells** [comment]
@NikCharlebois thank you following up and am working on it, however having issues with updating ReverseDSC module to support testing this. Opened an issue for that https://github.com/microsoft/ReverseDSC/issues/17

**Comment by adrianwells** [comment]
@NikCharlebois using the Dev branch (version 1.0.0.1048), it does not yet appear to be working however there is an errorlog that was created:

```
[2019/12/02 01:46:56]
{InvalidOperation}
System.Management.Automation.RuntimeException: You cannot call a method on a null-valued expression.
   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
   at <ScriptBlock>(Closure , FunctionContext )
   at System.Management.Automation.ScriptBlock.InvokeWithPipeImpl(ScriptBlockClauseToInvoke clauseToInvoke, Boolean createLocalScope, Dictionary`2 functionsToDefine, List`1 variablesToDefine, ErrorHandlingBehavior errorHandlingBehavior, Object dollarUnder, Object input, Object scriptThis, Pipe outputPipe, InvocationInfo invocationInfo, Object[] args)
   at System.Management.Automation.ScriptBlock.<>c__DisplayClass57_0.<InvokeWithPipe>b__0()
   at System.Management.Automation.Runspaces.RunspaceBase.RunActionIfNoRunningPipelinesWithThreadCheck(Action action)
   at System.Management.Automation.ScriptBlock.InvokeWithPipe(Boolean useLocalScope, ErrorHandlingBehavior errorHandlingBehavior, Object dollarUnder, Object input, Object scriptThis, Pipe outputPipe, InvocationInfo invocationInfo, Boolean propagateAllExceptionsToTop, List`1 variablesToDefine, Dictionary`2 functionsToDefine, Object[] args)
   at System.Management.Automation.ScriptBlock.InvokeUsingCmdlet(Cmdlet contextCmdlet, Boolean useLocalScope, ErrorHandlingBehavior errorHandlingBehavior, Object dollarUnder, Object input, Object scriptThis, Object[] args)
   at Microsoft.PowerShell.Commands.ForEachObjectCommand.ProcessRecord()
   at System.Management.Automation.CommandProcessor.ProcessRecord()
""
```

**Comment by adrianwells** [comment]
@NikCharlebois in some conditions the `Export-O365Configuration` cmdlet does prompt for a Destination Path. For example when only selecting "Admin Audit Log Config" under Office 365 or only selecting "Security & Compliance". Maybe it works only when excluding the Users and Groups data?

**Comment by NikCharlebois** [comment]
You can always try that. When you run it and it doesn't prompt for Destination path, what is the last Verbose message in the console that shows? E.g. "Extracting O365User..."

**Comment by adrianwells** [comment]
@NikCharlebois it stops here:

```
Extracting O365Group...
    - [1/20] Exchange Servers
    - [2/20] Administrative Department
    - [3/20] All Registered HW IDs
```

That last Azure AD group in that list, "All Registered HW IDs", is of a Dynamic type whereas all other groups are Assigned types. Maybe that is an issue?

Another item, found the module to exit the PowerShell terminal suddenly (crash) after running `Export-O365Configuration`, entering credentials and then pressing enter. That was proceeded with an `Unblock-File` cmdlet. Workaround seems to be not running the Unblock-File cmdlet.

**Comment by NikCharlebois** [comment]
I can't foresee the Unblocking of file having anything to do with this. A new version of the module has been released on the gallery last week. Can I ask you to try installing it:

Install-Module Office365DSC -Force

On a new environment (e.g. new VM) and try to replicate this behavior there as well? Thanks

**Comment by adrianwells** [comment]
@NikCharlebois the most current version (1.0.1.1395) on a "fresh" Windows OS seems to have the same issue with becoming the last Azure AD group in that list, "All Registered HW IDs", is of a Dynamic type whereas all other groups are Assigned types.

Not seeing an issue with crashing so that seems better with this version.

**Comment by rosskarchner** [comment]
I'm having the same symptom, but the cause may be different-- Fresh Windows 10 Home with all available updates, Office365DSC 1.0.1.1395. Everything appears to run fine, but I never get prompted for a location to save to.

I'm invoking with `Export-O365Configuration`, and I've tried setting -MaxProcesses (see the error below)


This gets written to 4468-O365DSC-ErrorLog:

```
[2019/12/20 10:06:02]
{MetadataError}
System.Management.Automation.ValidationMetadataException: The variable cannot be validated because the value 0 is not a valid value for the MaxProcesses variable.
   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.Interpreter.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.LightLambda.RunVoid1[T0](T0 arg0)
   at System.Management.Automation.PSScriptCmdlet.RunClause(Action`1 clause, Object dollarUnderbar, Object inputToProcess)
   at System.Management.Automation.PSScriptCmdlet.DoEndProcessing()
   at System.Management.Automation.CommandProcessorBase.Complete()
""
```

I'm

**Comment by rosskarchner** [comment]
I just tried on a Server 2019 instance (10.0.17763 Build 17763) and got the same result. Nothing but:

`Install-Module -Name Office365DSC -Force`
`Export-0365Configuration`.

It gets all the way to "Extracting TeamUser" and just silently stops, writing that same error as above to the log.

**Comment by NikCharlebois** [comment]
Folks, we believe we found the source of the issue. Can one of you, who was experiencing the issue, try out the latest codebase from GitHub and confirm that this has been fixed? Thanks

**Comment by rosskarchner** [comment]
Confession: I'm a total PS newbie, and searching has failed me. I haven't yet figured out how to install a module from a local checkout..

**Comment by rosskarchner** [comment]
OK, here's what I did. Installed Office365DSC with `Install-Module -Name Office365DSC -Force`, and then used `Install-ModuleFromGitHub` to grab the latest code from git.

I still don't get prompted for a location to save to, but the logged error is different:

```

**Comment by Worrelpa** [comment]
I tried this the export for the first time last night. New machine clean. Windows 10 home. Same issue. No prompt for results. everything else ran as expected. Tried numerous times, from the powershell command prompt and the ISE. Tried changing what data was selected to come back. No go. 

only error log messages I got were all similar to below. 

[2020/01/21 10:54:05]
{InvalidOperation}
System.Management.Automation.RuntimeException: Method invocation failed because [Microsoft.IdentityModel.Clients.ActiveDirectory.AuthenticationContext] does not contain a method named 'AcquireTokenByRefreshToken'.
   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.Interpreter.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.LightLambda.RunVoid1[T0](T0 arg0)
   at System.Management.Automation.PSScriptCmdlet.RunClause(Action`1 clause, Object dollarUnderbar, Object inputToProcess)
   at System.Management.Automation.PSScriptCmdlet.DoEndProcessing()
   at System.Management.Automation.CommandProcessorBase.Complete()
""
